client/src/context/socket.js

```diff
@@ -0,0 +1,7 @@
+import { createContext } from 'react';
+import socketIOClient from 'socket.io-client';
+
+const ENDPOINT = 'http://localhost:3000';
+const SocketContext = createContext(socketIOClient(ENDPOINT, { path: '/sockets' }));
```

Comment:
I believe this may autoConnect (unless you give the autoConnect: false) attribute. This can be dangerous since you want to control when connection happens, so you may want to wrap socketIOClient in a useEffect and do it only once the user is set in the UserContext